test/e2e/operator_test.go

```diff
@@ -1462,6 +1463,192 @@ func TestNetworkLoadBalancer(t *testing.T) {
 	}
 }
 
+func TestUniqueIdHeader(t *testing.T) {
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "uniqueid"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+	ic.Spec.HTTPHeaders = &operatorv1.IngressControllerHTTPHeaders{
+		UniqueId: operatorv1.IngressControllerHTTPUniqueIdHeaderPolicy{Name: "x-unique-id"},
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+	conditions := []operatorv1.OperatorCondition{
+		{Type: operatorv1.IngressControllerAvailableConditionType, Status: operatorv1.ConditionTrue},
+		{Type: operatorv1.LoadBalancerManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+		{Type: operatorv1.DNSManagedIngressConditionType, Status: operatorv1.ConditionFalse},
+	}
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, icName, conditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller deployment: %v", err)
+	}
+
+	service := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.InternalIngressControllerServiceName(ic), service); err != nil {
+		t.Fatalf("failed to get ingresscontroller service: %v", err)
+	}
+
+	echoPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"app": "unique-id-echo",
+			},
+			Name:      "unique-id-echo",
+			Namespace: deployment.Namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
```

Comment:
I've noticed more than one test in this repo at this point make use of this "echo" container spec.
I believe the same could be say about the curl client container.
Since the e2e test files in this repo keep getting longer and longer, I think it would be wise to try and re-use common test container specs (and any other common functionality, really) within the repo. Thoughts? Definitely doesn't need to happen in this PR, so perhaps we should open a Jira ticket to track this improvement?

Reply:
Latest push does some refactoring to reduce duplicate (still more could be done though).

Reply:
Great, thanks for that!